src/libraries/System.Security.Cryptography.ProtectedData/src/System/Security/Cryptography/ProtectedData.cs

```diff
@@ -388,16 +390,16 @@ private static bool TryProtectOrUnprotect(
                             return false;
                         }
 
-                        new Span<byte>(outputBlob.pbData.ToPointer(), length).CopyTo(outputSpan);
+                        interopSpan.CopyTo(outputSpan);
                         bytesWritten = length;
                         outputData = null;
                         return true;
                     }
                     finally
                     {
-                        if (outputBlob.pbData != IntPtr.Zero)
+                        if (!interopSpan.IsEmpty)
```

Comment:
Is this correct? What happens if something unprotects to a zero-length value? Is there still something that needs to be freed?

Reply:
Hrm... that's a good point... I guess something did _technically_ allocate some zero length thing somewhere that might mess something up. I'll put it back to the IntPtr comparison.

Reply:
I was trying to contain usages of the raw pointers since I had the nice span already, but in this case, I think this specific case isn't a good place for that.